### PR TITLE
Add semantic checks to avoid NPE in Mapper.

### PR DESCRIPTION
**Pull request by zouyonghao**
Signed-off-by: zouyonghao <yonghaoz1994@gmail.com>

---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #433 |
| Primary OS tested on | Ubuntu, |
| Robotic platform tested on | gazebo simulation turtlebot |

---

## Description of contribution in a few bullet points

* Add semantic checks to avoid NPE in Mapper.

### CODE REVIEW

File: src/laser_utils.cpp [hunk 1]
```diff
@@ -125,6 +125,13 @@ karto::LaserRangeFinder * LaserAssistant::makeLaser(const double & mountingYaw)
   }
   node_->get_parameter("max_laser_range", max_laser_range);
 
+  if (max_laser_range <= 0) {
```
Comment (SteveMacenski) [inline src/laser_utils.cpp:128]: Is this a real issue you really ran into? This seems like its from a poorly formed laser scan message and it should be fixed by the publisher

File: src/slam_mapper.cpp [hunk 2]
```diff
@@ -207,6 +207,12 @@ void SMapper::configure(const rclcpp::Node::SharedPtr & node)
       correlation_search_space_dimension);
   }
   node->get_parameter("correlation_search_space_dimension", correlation_search_space_dimension);
+  if (correlation_search_space_dimension <= 0) {
```
Comment (SteveMacenski) [inline src/slam_mapper.cpp:210]: Did you catch all of the error conditions in settings?

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: src/laser_utils.cpp (hunk 1 above)
Comment (zouyonghao) [reply src/laser_utils.cpp:128]: I just consider it as a parameter a user can set, and it seems it is not related to scan message...

File: src/slam_mapper.cpp (hunk 2 above)
Comment (zouyonghao) [reply src/slam_mapper.cpp:210]: I catch all error conditions that can cause #433

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**